### DISCUSSION TITLE
AstroNvim v2.0 (Neovim v0.8) Nightly Release

### DISCUSSION THREAD

**Discussion by mehalter** [AstroNvim/AstroNvim#1020] [Announcements]
With the official stable release of [Neovim v0.8](https://github.com/neovim/neovim/releases/tag/v0.8.0), we are excited to announce that AstroNvim v2.0 is now in the nightly release.

Starting with AstroNvim v2.0 we are going to be making the `stable` release channel the default update channel for AstroNvim. This means that the `main` branch on the git repository will now only be tracking the official tagged releases and all nightly development will be moved to a dedicated `nightly` branch. If you update to Neovim v0.8 and want to try the new AstroNvim v2.0 you will have to update your `updater` settings in your `user/init.lua` (or `user/updater.lua` if you like to split up your user confgiuration). You can use the following options to be able to pull the new version with `:AstroUpdate`:

```lua
updater = {
  branch = "nightly",
}
```

*Please Note:* That with these new breaking changes to Neovim, AstroNvim 1.X will not be supported on Neovim v0.8 and AstroNvim  2.X will not be supported below Neovim v0.8. If you are need to make sure that your AstroNvim does not update when the new release is made, you can use the following updater settings to make sure you stay on the last `v1.10.0` release:

```lua
updater = {
  channel = "stable",
  version = "v1.10.0",
}
```

This update is coming with a fairly lengthy list of breaking changes and in preparation we have been hard at work making a [Migration Guide](https://astronvim.github.io/nightly/Configuration/v2_migration) to help users update their user configurations to be compatible with AstroNvim v2.0. Over the next week as we let users test AstroNvim v2.0 before the stable release, it would also be great to get feedback from the community on the migration guide to make sure it's top notch for the rest of the users that will be forced to use it as AstroNvim v2.0 becomes stable.

[**Migration Guide from v1.X to v2.0**](https://astronvim.github.io/nightly/Configuration/v2_migration)

Thank you to everyone that will help us test out this new release and I hope the migration is not too tedious!

**Comment by Noxdor** [comment]
I just took the time to read through the documentation and I think you guys did a great job with 2.0! I love literally every single change (though some don't effect me / are not important to me), but the ones that do, they really make it even more enjoyable to use AstroNvim! I feel like everything became much more stable, specially since you guys finally removed the `Esc` from the floating terminal and things like `lazygit`, that is a huge improvement imo! And I love the `<Leader> u` menu!

Three questions I had after reading the migration guide:

1. Since ToggleTerm is now not anymore on `Ctrl + \` I run into a problem: Imagine I am using that terminal window to run a script that keeps watching files or runs a server (something like `npm run serve`). How do I skip out of the floating terminal to continue working while letting the terminal run in the background? I can't use `<Leader> t f` like I did to open it.
2. How can I change the default system monitor (now `bottom`). I am really enjoying `btop` and would like to switch to it!
3. The open file tabs were removed, instead I can see `File`, `Bufs` and `Git` above the `neo-tree` window. That looks interesting and promising, but I couln't figure out how to switch to `Bufs` for example, where I guess I could find the other opened files 😅

I hope this pretty unstructured rant is helping with the migration guide! I am using AstroNvim for a while now, but without being very deep into Lua, the Nvim ecosystem and plugin development, so please see my comment as coming from a casual user that is mainly focused on using AstroNvim in his daily work and endeavours. Appreciate so much what you guys are creating here and so exciting to see this project go further and further!

**Reply by mehalter** [reply]
Thanks for this feedback @Noxdor ! I can answer these questions:

1. This is a good point, I think we can add a new binding to the terminal menu for toggling. I just pushed an update to add `<leader>t<cr>` to have a good way to get this functionality back without breaking the default binding of `ctrl + \`

2. You can do this by simply overriding the mapping:

```lua
return {
  mappings = {
    n = {
      ["<leader>tt"] = { function() astronvim.toggle_term_cmd "btop" end, desc = "ToggleTerm btop" },
    },
  },
}
```

3. Great question! These can be switched with `<` and `>`, you can also use `?` to browse the mappings for `neo-tree`. But also they can be clicked with the mouse :eyes: (Also this wasn't explicit in the migration guide but will be described in the release notes later this week, but most of our statusline elements are now clickable as well.... (clicking the git branch in the statusline will let you switch branches, clicking the diagnostic icons will bring up all the diagnostics in telescope, clicking git change icons will open a list of the chunks in telescope, clicking the language server names will open the LSP information, and clicking the breadcrumb elements in the `winbar` will navigate up the document to the clicked element :eyes:

**Reply by Noxdor** [reply]
1. Either I am doing it wrong, or this doesn't work for me 😬 Since `<leader>` is the spacebar, this is just causing an input inside the terminal, instead of toggling it. I guess it really does have to be some combination with `ctrl`?
2. Works like a charm and looks awesome 😍 
3. Awesome, thank you! Btw I just realised that you didn't remove the open file tabs on the top like I said, they are just not working properly, I guess because it's still nightly? For example, I had to first use `open_tabnew` (t) in `nvim_tree` to see them again:

Bofore: 
![image](https://user-images.githubusercontent.com/64439978/193669538-be9955d5-4a87-4c11-abf8-689784cd8035.png)

After: 
![image](https://user-images.githubusercontent.com/64439978/193669494-f2528792-bd31-4067-9f43-b518cdd4e307.png)

I am sure you have that issue on the radar, just wanted to tell you guys if not :)

While I am at it, what I was always missing was a way to rearrange the open files/buffers. So often I have a situation where I have to switch quickly and often between two or more files, and if I opened other files in between, it is costing time to skip over them (talking about `<Shift> + H, <Shift> + L` here). Is there already the functionality to reorder these windows? If not I think it would be an awesome feature :)

**Reply by queue-tip** [reply]
The default [mappings for Bufferline](https://astronvim.github.io/Basic%20Usage/mappings#bufferline)  are ">b" and "<b" to move buffers right and left respectively.

**Reply by mehalter** [reply]
@Noxdor I have pushed an update to fix the tabline issue! This was a breaking change that slipped through the cracks

**Reply by mehalter** [reply]
@Noxdor I'm not sure of a good binding to include out of the box with control without overriding useful features. Do you have any recommendations? Also feel free to add this as a mapping in your user config to something that does fit your workflow as well

**Reply by Noxdor** [reply]
Update works like a charm! Thanks!

I honestly don't know enough bindings to know what would be a good option and what would break stuff (I don't even know what `<Ctrl> + \` broke 😂). I'd have no problem setting this myself, but imo this is a very integral feature of AstroNvim. If I can't toggle the floating terminal off while a process is running in there, the usefulness of that terminal is drastically reduced. For that reason I think it should have already a default keybinding that makes this work! It's your call of course after all, I am just giving my thought :) Maybe the `<Alt>` key could be used? Or is there a way to have a keybinding only with the right `<Ctrl>` key?

**Reply by mehalter** [reply]
Actually @Noxdor I found that `ctrl-'` is not bound by default and it's a very comfortable binding ! I changed the binding to that instead of `space+t+enter`

**Reply by Noxdor** [reply]
Awesome! But I am not sure what exactly is this combination, is it `ctrl` and `'` pressed at the same time? If so, it isn't working for me 😅 What **is** working (either this is what you actually meant now or I discovered it by accident) is `ctrl` + `Backspace` 🤔 I can now open the floating terminal witth `<Leader> t f` and close it with `ctrl + Backspace`. The terminal process is not ended and the running script is still there when I reopen it with `<Leader> t f`.

**Reply by Zedai00** [reply]
sumneko_lua and stylua both try to format the config file and the box keeps appearing how to set it so only one thing formats it.

**Comment by natematykiewicz** [comment]
I tried updating to the nightly after I accidentally updated to Neovim 0.8.

I set the channel to nightly in `~/.config/nvim/lua/user/init.lua`:
```lua
local config = {
  updater = {
    remote = "origin", -- remote to use
    channel = "nightly", -- "stable" or "nightly"
    version = "latest", -- "latest", tag name, or regex search like "v1.*" to only do updates before v2 (STABLE ONLY)
    branch = "main", -- branch name (NIGHTLY ONLY)
    commit = nil, -- commit hash (NIGHTLY ONLY)
    pin_plugins = nil, -- nil, true, false (nil will pin plugins on stable only)
    skip_prompts = false, -- skip prompts about breaking changes
    show_changelog = true, -- show the changelog after performing an update
    auto_reload = false, -- automatically reload and sync packer after a successful update
    auto_quit = false, -- automatically quit the current session after a successful update
    -- remotes = { -- easily add new remotes to track
    --   ["remote_name"] = "https://remote_url.come/repo.git", -- full remote url
    --   ["remote2"] = "github_user/repo", -- GitHub user/repo shortcut,
    --   ["remote3"] = "github_user", -- GitHub user assume AstroNvim fork
    -- },
  },
```

Ran `:AstroUpdate`. Then quit. Then opened Neovim. Then `:PackerSync`. Reloaded Neovim again. I still have no syntax highlighting without setting a `ft`.

`:AstroUpdate` says "No updates available"
`:AstroVersion` says "Version: v1.10.0" in the top right corner

Am I missing a step? I did convert everything from https://astronvim.github.io/nightly/Configuration/v2_migration as well.

**Reply by natematykiewicz** [reply]
I overlooked that the _branch_ in the original code example in this discussion was nightly. I thought it was _channel_ nightly. **_Both_** the channel and branch need to be "nightly". Rereading (for a 4th time now) I spotted the difference.

It felt confusing because the second example is channel/version. So I thought the first one was channel.

**Reply by Fenkiou** [reply]
Many thanks 🙏 

The first post indicates `branch = nightly` and the documentation indicates `channel = nightly`, needed both too.

**Reply by natematykiewicz** [reply]
Yeah, I think it would help if this post were updated to say:

```lua
updater = {
  channel = "nightly",
  branch = "nightly",
}
```

**Comment by A-Lamia** [comment]
Hey awesome update, 
how would we go about separating the the heirline examples in to separate files ?

**Reply by mehalter** [reply]
Hey @A-Lamia you can just put whatever you put in the `heirline` key into `user/heirline.lua`. check out the table of all possible options: https://astronvim.github.io/Configuration/config_options

**Reply by A-Lamia** [reply]
Does this also apply to the plugin override ?

**Reply by A-Lamia** [reply]
file `user/heirline.lua` code
```lua
return {
  separators = {
    left = { "", " " }, -- separator for the left side of the statusline
    right = { " ", "" }, -- separator for the right side of the statusline
  },
  -- add new colors that can be used by heirline
  colors = {
    blank_bg = "#5c6370",
    file_info_bg = "#3e4452",
    nav_icon_bg = "#89b06d",
    folder_icon_bg = "#ec5f67",
  },
}
```

is this correct ?

**Comment by yehorb** [comment]
Remark for everyone upgrading - if you encounter any instabilities or weird plugin behavior - try doing the "clean install" - e.g. remove the `nvim-data` directory altogether, and let `AstroNvim` and `packer` reinstall and reconfigure all of the plugins. For me, this solved `tresitter` highlighting issues (which seem pretty common after 0.7.2 -> 0.8.0 upgrade overall) and weird `nvim-cmp` behavior.

**Reply by nixyhustle** [reply]
Where would one typically find this nvim-data directory? Fairly new to all this and I am trying to get syntax highlighting working again. Another fix I saw was downgrading the nvim version, but I have no problem trying a "clean install" as you mentioned.

**Reply by yehorb** [reply]
Use the command `:echo expand("data")` to get the absolute path to the data directory.
